File: CodingSamples/Java/Generics/TypeParamTest2/Program2.java
class Program {

	//private static void printStack(SimpleStack<? extends Object> store) {
	private static void printStack(SimpleStack<?> store) {
		while(!store.empty())
			System.out.println(store.pop());
		//store.push("Jack");
	}

	public static void main(String[] args) {
		SimpleStack<String> a = new SimpleStack<String>();
		a.push("Monday");
		a.push("Tuesday");
		a.push("Wednesday");
		a.push("Thursday");
		a.push("Friday");
		printStack(a);
		System.out.println("-----------------");
		SimpleStack<Double> b = new SimpleStack<>(); 
		b.push(23.1);
		b.push(45.2);
		b.push(56.3);
		b.push(31.4);
		printStack(b);
		System.out.println("-----------------");
		SimpleStack<Object> c = new SimpleStack<>();
		c.push("Sunday");
		c.push(97.5);
		c.push(1234);
		printStack(c);
	}
}

File: CodingSamples/DotNet/Generics/TypeParamTest2/DemoApp/Program2.cs
class Program
{
    private static void PrintStack(IPoppable<object> store)
    {
        while(store.Count > 0)
            Console.WriteLine(store.Pop());
    }

    public static void Main(string[] args)
    {
        SimpleStack<string> a = new SimpleStack<string>();
        a.Push("Monday");
        a.Push("Tuesday");
        a.Push("Wednesday");
        a.Push("Thursday");
        a.Push("Friday");
        //a.Push(1.23);
        PrintStack(a);
        Console.WriteLine("-----------------------");
        SimpleStack<Interval> b = new SimpleStack<Interval>();
        b.Push(new Interval(5, 31));
        b.Push(new Interval(7, 42));
        b.Push(new Interval(6, 53));
        b.Push(new (3, 14));
        PrintStack(b);
        Console.WriteLine("-----------------------");
        SimpleStack<object> c = new SimpleStack<object>();
        c.Push("Sunday");
        c.Push(1.23);
        c.Push(new Interval(4, 25));
        PrintStack(c);
        //SimpleStack<double> d = new SimpleStack<double>();
        //PrintStack(d);
    }
}
